### PR TITLE
Api endpoints

### PR DESCRIPTION
**Pull request by tkmct**
Add API Endpoints for Coin transfer. Add TransactionFilter for querying transactions

### CODE REVIEW

File: ovm/src/db/transaction_db.rs [hunk 1]
```diff
@@ -43,4 +45,68 @@ where
...
+        &self,
+        transaction_filter: TransactionFilter,
+    ) -> Result<Vec<Transaction>, AbiError> {
+        let from_block = transaction_filter.get_from_block();
+        let to_block = transaction_filter.get_to_block();
+        let range = transaction_filter.get_range();
+        let mut result = vec![];
+        for i in from_block..=to_block {
+            let txs = self.get_transactions(i, range)?;
+            println!("{:?}", txs);
```
Comment (syuhei176) [inline ovm/src/db/transaction_db.rs]: for debug?

File: core/src/data_structure/transaction.rs [hunk 2]
```diff
@@ -82,6 +83,7 @@ pub struct Transaction {
     range: Range,
     parameters: Bytes,
     signature: Bytes,
+    metadata: Metadata,
```
Comment (syuhei176) [inline core/src/data_structure/transaction.rs]: I hope we can store any data as metadata in the future. It's ok for now.

Comment (tkmct) [reply core/src/data_structure/transaction.rs]: Right, I wanted it in Bytes but for ease, I put that in specific type.

File: ovm/src/db/transaction_db.rs (hunk 1 above)
Comment (tkmct) [reply ovm/src/db/transaction_db.rs]: Ooops

**Pull request merged**